### DISCUSSION TITLE
fields not working in .find() ?

### DISCUSSION THREAD

**Discussion by esinencodorini** [mikro-orm/mikro-orm#6142] [General]
### Describe the bug

Hi, first of all thanks for the awesome framework.

when i use repository on all of my entities i use ```this.find(where, options)```, where options is ```{fields: ['uuid', 'roles']}``` - but instead I get the entire entity, why ?

### Reproduction

in nestjs

1. Define an CustomEntityRepository extending the mikro orm EntityRepository<CustomEntity>
2. within a service method inject the repository
3. find any entity using injected repository .find({where...}, {fields: [... desired fields]})
4. expected output -> the entity loads fields not mentioned in the fields option

### What driver are you using?

postgresql

### MikroORM version

6.3.12

### Node.js version

20.18.0

### Operating system

windows / linux

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
> Reproduction

This is not a [reproduction](https://github.com/mikro-orm/reproduction), hard to help without seeing what you are doing. What you described is covered in many tests.

A common problem is that your entity is already loaded in the context, e.g. via some auth middleware.
